pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -964,4 +963,128 @@ self: super: {
   # https://github.com/danidiaz/pipes-transduce/issues/2
   pipes-transduce = super.pipes-transduce.override { foldl = self.foldl_1_1_6; };
 
+  # Haste stuff:
+  # ============
+
+  # $ nix-shell -p "haskellPackages.ghcWithPackages (self: with self; [haste-cabal-install haste-compiler])" -p nodejs
+  # $ haste-cabal update # Optional but will often fix "haste-cabal: Could not resolve dependencies" during haste-boot
+  # $ haste-boot
+  # $ echo 'module Main where main = putStrLn "Hello world"' > hello-world.hs
+  # $ hastec --onexec hello-world.hs
+  # $ node hello-world.js
+  # Hello world
+
+  # Haste requires its own patched up version of Cabal that's not on hackage
+  haste-Cabal = self.callPackage
```

Comment:
Would you mind putting those expressions into separate files instead of inlining them here in their entirety? That file is rather long already. Check out the overrides for `cabal2nix` that do the same thing for an example.